tempodb/backend/cache/memcached/memcached.go

```diff
@@ -0,0 +1,60 @@
+package memcached
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+
+	cortex_cache "github.com/cortexproject/cortex/pkg/chunk/cache"
+	"github.com/grafana/tempo/tempodb/backend/cache"
+)
+
+type Config struct {
+	ClientConfig cortex_cache.MemcachedClientConfig `yaml:",inline"`
+
+	TTL time.Duration `yaml:"ttl"`
+}
+
+type Cache struct {
```

Comment:
Let's name this `Client`. That way when we create the client the signature will be `memcache.NewClient()`., which also helps reduce confusion. Similarly for Redis.